python/pyspark/sql/connect/dataframe.py

```diff
@@ -138,6 +138,41 @@ def __init__(
         # by __repr__ and _repr_html_ while eager evaluation opens.
         self._support_repr_html = False
         self._cached_schema: Optional[StructType] = None
+        self._cached_remote_relation_id: Optional[str] = None
+
+    def __del__(self) -> None:
```

Comment:
So no Python GC expert here. I am assuming some system thread is doing this work. Is it wise to execute an RPC from there?